### PR TITLE
Feature/form

### PR DESCRIPTION
**Pull request by RileyManda**
I have completed the task of adding a form  to the Blog Application according to the stipulated [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/forms_project_v1_1.md): 🌱 

# Implementations: 

### Views

> Created a form that Creates a Post on behalf of the current_user 
> Created a comment on behalf of the current_user 
> Added Feature to enable users to like Posts

 In addition I have ensured the following general requirements:
 
[Linters](https://github.com/RileyManda/blog-app/actions/workflows/linters.yml)

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

Please kindly review my implementation.

Thank you 🙏🏻

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -10,4 +10,32 @@ def show
...
+      likes_counter: 0
+    )
+
+    if @post.save
+      flash[:success] = 'Post created successfully!'
+      redirect_to user_posts_path(@user)
+    else
+      flash[:error] = 'Error creating the post.'
+      render 'new'
+    end
```
Comment (iambenkis) [inline app/controllers/posts_controller.rb]: - [ ] Good job creating the Post create method, note one imported thing : the post should e created on the behalf of the `current_user` which is f the currently logged-in user 👍 

```ruby
.
.
.
def create
      @post = @current_user.posts.new(post_params)
.
.
.
```

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @RileyManda 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  Add new post form 
✔️ Add new comment form 
✔️ Add life feature

## Suggested changes

- [ ] [**OPTIONAL**] Good job adding forms  👍 , I will suggest you to provide likes in your file that allow user to navigate through different pages , for example user should see a button that redirect to the new post page , a button  that redirect to all the posts 👍 .

- A button to redirect to the page below 
Capture

Capture1


_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by RileyManda** [comment]
I have completed the task of updating post controller create method as requested by the reviewer: 🌱 

# Implementations: 

### Controllers
> Set the author of a new post to the current user in PostsController#create

### Views [Optional]

> Implemented a create post button


 In addition I have ensured the following general requirements:
 
[Linters](https://github.com/RileyManda/blog-app/actions/workflows/linters.yml)

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review my implementation.

Thank you 🙏🏻

**Comment by RileyManda** [comment]
I have completed the task of updating post controller create method as requested by the reviewer: 🌱 

# Iterations: 

### Controllers
> Set the author of a new post to the current user in PostsController#create

### Views [Optional]

> Implemented a create post button


 In addition I have ensured the following general requirements:
 
[Linters](https://github.com/RileyManda/blog-app/actions/workflows/linters.yml)

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review my implementation.

Thank you 🙏🏻

**Comment by RileyManda** [comment]
I have completed the task of updating post controller create method as requested by the reviewer: 🌱 

# Iterations: 

### Controllers
> Set the author of a new post to the current user in PostsController#create

### Views [Optional]

> Implemented a create post button


 In addition I have ensured the following general requirements:
 
[Linters](https://github.com/RileyManda/blog-app/actions/workflows/linters.yml)

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review my implementation.

Thank you 🙏🏻

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @RileyManda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] All application forms work as expected :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark:  
- [x] Implemented all the changes from the previous reviewer :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    

## Optional suggestions
[OPTIONAL] Good job adding forms 👍 , I would like to suggest that you provide links in your application that allow a user to navigate through different pages, for example, a user should be able to `click` on a `post` to open the `show` post page, a user should also be able to navigate the `users` show page by clicking on a particular user. Please implement this in order to improve your application. 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**